addons/html_editor/static/src/main/media/image_crop_plugin.js

```diff
@@ -42,19 +35,25 @@ export class ImageCropPlugin extends Plugin {
             return;
         }
 
-        this.imageCropProps.media = selectedImg;
-
-        const onClose = () => {
-            registry.category("main_components").remove("ImageCropping");
-        };
-
-        const onSave = () => {
-            this.dependencies.history.addStep();
-        };
-
         registry.category("main_components").add("ImageCropping", {
             Component: ImageCrop,
-            props: { ...this.imageCropProps, onClose, onSave, document: this.document },
+            props: {
+                media: selectedImg,
+                onClose: () => {
+                    registry.category("main_components").remove("ImageCropping");
+                },
+                onSave: async (newDataset) => {
+                    // todo: should use the mutex if there is one?
+                    const updateImageAttributes =
+                        await this.dependencies.imagePostProcess.processImage(
+                            selectedImg,
+                            newDataset
+                        );
+                    updateImageAttributes();
+                    this.dependencies.history.addStep();
+                },
+                document: this.document,
```

Comment:
mutex ?

Reply:
I adapted the previous code that used the method `applyModification` as the equivalent now is `processImage`.
While adapting the code, I noticed that the mutation made by `updateImageAttributes` will not be mutexified with the `OperationPlugin` and will corrupt the history in some circumstance.
In practice, I don't think this method will be called in the `html_builder` so we're safe for now.

Reply:
todo: ensure no mutation in load